sql/src/main/java/io/crate/metadata/settings/CrateSettings.java

```diff
@@ -287,11 +288,46 @@ Settings settings() {
     private Map<String, ReferenceImplementation> buildReferenceTree() {
         Map<String, ReferenceImplementation> referenceMap = new HashMap<>(BUILT_IN_SETTINGS.size());
         for (CrateSetting crateSetting : BUILT_IN_SETTINGS) {
+            if (crateSetting.isGroupSetting()) {
+                Map<String, Settings> settingsMap = initialSettings.getGroups(crateSetting.getKey(), true);
+                for (Map.Entry<String, Settings> entry : settingsMap.entrySet()) {
+                    buildGroupSettingReferenceTree(crateSetting.getKey(), entry, referenceMap);
+                }
+            }
             buildReferenceTree(referenceMap, crateSetting);
         }
         return referenceMap;
     }
 
+    @VisibleForTesting
+    void buildGroupSettingReferenceTree(String prefix, Map.Entry<String, Settings> settingEntry, Map<String, ReferenceImplementation> referenceMap) {
+        Map<String, String> settingsMap = settingEntry.getValue().getAsMap();
+
+        //this is a nested setting
+        if (!settingsMap.isEmpty()) {
+            //we need to build the reference tree for the current setting
+            buildReferenceTree(referenceMap,
+                CrateSetting.of(Setting.groupSetting(prefix + settingEntry.getKey() + ".",
+                    Setting.Property.NodeScope),
+                    DataTypes.OBJECT));
+            //build the reference tree for every child setting
+            for (Map.Entry<String, String> entry : settingsMap.entrySet()) {
+                String nestedPrefix = prefix + settingEntry.getKey() + "." + entry.getKey();
+
+                buildReferenceTree(referenceMap,
+                    CrateSetting.of(Setting.simpleString(nestedPrefix,
+                        Setting.Property.NodeScope),
+                        DataTypes.STRING));
+            }
+        } else {
+            //this is a simple setting 
+            buildReferenceTree(referenceMap,
```

Comment:
@seut If a setting is defined like this ```{"global": {"nested":{"key-a":"value-a", "key-b":"value-b"}, "enabled": true}}```

and 

```
        CrateSetting<Settings> TEST_SETTING = CrateSetting.of(
            Setting.groupSetting("global.nested.",
                Setting.Property.Dynamic,
                Setting.Property.NodeScope),
            DataTypes.OBJECT
        );
```

getGroups() does not contain the key:``enabled``

i guess for this case, we can define the group setting as ``global.``
because having ``global.nested.enabled`` as a separate CrateSetting, will result in a conflict..

anyways it's removed